src/ssl.rs

```diff
@@ -351,6 +388,12 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
     pub fn try_read(&self, data_out: &mut BytesMut) -> PollResult<usize> {
         let buf = data_out.spare_capacity_mut();
 
+        // SAFETY: [`wolfSSL_read`][0] ([also][1]) expects valid pointer to `WOLFSSL` and since the `WOLFSSL` struct
+        // can be used in multiple threads based on [`Library design`][2], protected by a mutex lock
```

Comment:
This ought to talk about the validity of `buf` here and the corresponding length, that's the most important thing from a safety point of view I think.